elements/lisk-transactions/src/utils/verify.ts

```diff
@@ -54,72 +53,89 @@ export const verifyMinRemainingBalance = (
 	return undefined;
 };
 
-export interface VerifyMultiSignatureResult {
-	readonly status: MultisignatureStatus;
-	readonly errors: ReadonlyArray<TransactionError>;
-}
-
-const isMultisignatureAccount = (account: Account): boolean =>
+export const isMultisignatureAccount = (account: Account): boolean =>
 	!!(
 		(account.keys.mandatoryKeys.length > 0 ||
 			account.keys.optionalKeys.length > 0) &&
 		account.keys.numberOfSignatures
 	);
 
-export const verifyMultiSignatures = (
+export const validateKeysSignatures = (
+	keys: readonly string[],
+	signatures: readonly string[],
+	transactionBytes: Buffer,
+) => {
+	const errors = [];
+	// tslint:disable-next-line: prefer-for-of no-let
+	for (let i = 0; i < keys.length; i += 1) {
+		const { error } = validateSignature(
+			keys[i],
+			signatures[i],
+			transactionBytes,
+		);
+
+		if (error) {
+			errors.push(error);
+		}
+	}
+
+	return errors;
+};
+
+export const verifyMultiSignatureTransaction = (
 	id: string,
 	sender: Account,
 	signatures: ReadonlyArray<string>,
 	transactionBytes: Buffer,
-): VerifyMultiSignatureResult => {
-	if (!isMultisignatureAccount(sender) && signatures.length > 0) {
-		return {
-			status: MultisignatureStatus.FAIL,
-			errors: [
-				new TransactionError(
-					'Sender is not a multisignature account',
-					id,
-					'.signatures',
-				),
-			],
-		};
+) => {
+	const errors = [];
+	if (signatures.length === 0) {
+		const error = new TransactionError(
+			'Transaction has empty signatures',
+			id,
+			'.signatures',
+		);
+
+		return [error];
 	}
 
-	if (!isMultisignatureAccount(sender)) {
-		return {
-			status: MultisignatureStatus.NONMULTISIGNATURE,
-			errors: [],
-		};
+	const { mandatoryKeys, optionalKeys, numberOfSignatures } = sender.keys;
+	const numMandatoryKeys = mandatoryKeys.length;
+	const nonEmptySignaturesCount = signatures.filter(k => k.length !== 0).length;
+
+	if (nonEmptySignaturesCount !== numberOfSignatures) {
```

Comment:
why do we need this check? `signatures.length !== numMandatoryKeys + optionalKeys.length`
in what case this will occur?

Reply:
I don't think there is any validation regards to the length of the signatures. 
ie) empty element was not injected property, it will throw
```
mandatory: 1, optional: 3, required signature 2
signatures: ['validPK', 'validOptionalForPosition4']
```